File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/GameUniverseToUI.java
/**
 * @file GameUniverseToUI.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representación ToUI de GameUniverse
 */
public class GameUniverseToUI {
    SpaceStationToUI currentStation;
    EnemyToUI currentEnemy;
    
    /**
     * @brief Constructor con parámetros
     * @param station Instancia de la clase SpaceStation
     * @param enemy Instancia de la clase EnemyStarShip
     */
    GameUniverseToUI(SpaceStation station,EnemyStarShip enemy) {
        currentStation = station.getUIversion();
        currentEnemy= enemy.getUIversion();
    }

    /**
     * @brief Consultor de la estación actual
     * @return currentStation
     */
    public SpaceStationToUI getCurrentStation() {
        return currentStation;
    }

    /**
     * @brief Consultor del enemigo actual
     * @return currentEnemy
     */
    public EnemyToUI getCurrentEnemy() {
        return currentEnemy;
    } 
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/GameUniverseToUI.rb
#encoding:utf-8

# File: GameUniverseToUI.rb
# Author: Profesor

module Deepspace

# Brief: Representación ToUI de GameUniverse
class GameUniverseToUI 
  
  # Consultores
  attr_reader :currentStation, :currentEnemy
      
  # Brief: Constructor con parámetros
  # Param station: Instancia de la clase SpaceStation
  # Param enemy: Instancia de la clase EnemyStarShip
  def initialize (station,enemy) 
      @currentStation = station.getUIversion()
      @currentEnemy= enemy.getUIversion()
  end

  public
  
  # Brief: Función para representar una instancia de la clase GameUniverseToUI
  #        en un string
  # Return: out String que representa una instancia de la clase GameUniverseToUI
  def to_s
    out = "Estación actual -> #{@currentStation}\n"
    out +="Enemigo actual --> #{@currentEnemy}"
    return out
  end
    
end # Class

end # Module